controllers/nodemaintenance_controller.go

```diff
@@ -148,8 +148,11 @@ func (r *NodeMaintenanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return emptyResult, nil
 	}
 
-	err = initMaintenanceStatus(ctx, nm, drainer, r.Client)
-	if err != nil {
+	if err = initMaintenanceStatus(ctx, nm, drainer); err != nil {
+		r.logger.Error(err, "Failed to initalize NodeMaintenance status")
+		return r.onReconcileError(ctx, nm, drainer, err)
+	}
+	if err = r.Client.Status().Update(ctx, nm); err != nil {
```

Comment:
do we need the update here? 🤔

Reply:
We want to update the status after it is initialized successfully (no error from line 151)

Reply:
> We want to update the status after it is initialized successfully (no error from line 151)

Absolutely.
But follow the code.
We already update the status when leave the Reconcile method, either at the end, or in `onReconcileError`.
I don't think we need it here as well.
WDYT?

Reply:
I agree it won't affect NMO functionality, but since we won't update the CR with the phase, etc., thus the user won't be able to view the CR status until there is a failure (e.g., from drain timeout) or a success. 

It may not be a nice user experience. Consider that a user can use evictionTimeout of 1 minute (by https://github.com/medik8s/node-maintenance-operator/pull/122), and in case of success, the CR status won't be updated (in the worst scenario) until one minute passed...

Reply:
I like your POV of saving API calls but I think this one should be kept, unless I am missing something here.

Reply:
fair point about about the long eviction update 👍🏼